autometacal/python/datasets/galaxies.py

```diff
@@ -0,0 +1,173 @@
+  Galaxy shapes follow a bivariate normal distribution centered in zero.
+  
+  Args:
+    g_range: galaxy shapes go from -g_range and + g_range in each g1, g2 
+    g_scatter: galaxy shapes scatter
+    flux: galaxy flux (counts)
+    pixel_scale: intended pixel scale in arcsec^2/pixel
+    stamp_size: size in pixels of the NxN resulting image
+    method: galsim drawing method
+    interpolator: galsim interpolation used to draw the image on the pixel grid.
+  Returns:
+    g1, g2: galaxy shape parameters
+    gal_image.array: numpy array that represents galaxy image
+  
+  """
+
+  defaults = {'g_range' : 0.8,        #elipticity
+              'g_scatter' : 0.25,     #
+              'mean_radius': 3.0,     #size
+              'scatter_radius': 0.1,  #
+              'psf_beta': 5,          #psf
+              'psf_fwhm': 0.5,        #
+              'mean_snr': 200,        #snr
+              'scatter_snr': 20,      #
+              'flux' : 1e5,           #flux
+              'pixel_scale' : 0.2,    #
+              'stamp_size' : 50,      #
+              'method' : "no_pixel"   #
+             }
+
+  defaults.update(kwargs)
+  
+  #ellipticity range
+  a, b = (-defaults['g_range'] - 0) / defaults['g_scatter'], (defaults['g_range'] - 0) / defaults['g_scatter']
+  g1=g2=1
+  
+  #select ellipticities, ensure g<=1
+  while g1**2+g2**2>1:
+    g1 = truncnorm.rvs(a, b, loc=0, scale=defaults['g_scatter'])
+    g2 = truncnorm.rvs(a, b, loc=0, scale=defaults['g_scatter'])
+    
+  re = norm.rvs(3, 0.1)
+  
+  #very simple galaxy model
+  gal = galsim.Exponential(flux=defaults['flux'] ,
+                           half_light_radius=re)
+  
+  #apply shear
+  gal = gal.shear(g1=g1,g2=g2)
+    
+  
+  #create constant psf
+  psf = galsim.Moffat(beta=defaults['psf_beta'], 
+                      fwhm=defaults['psf_fwhm'])
+  
+  #draw galaxy before convolution
+  true_gal_image = gal.drawImage(nx=defaults['stamp_size'],
+                            ny=defaults['stamp_size'],
+                            scale=defaults['pixel_scale'],
+                            method=defaults['method'])
+  
+  #convolve galaxy and psf
+  gal = galsim.Convolve([gal,psf])
+  
+  #draw psf image
+  psf_image = psf.drawImage(nx=defaults['stamp_size'],
+                            ny=defaults['stamp_size'],
+                            scale=defaults['pixel_scale'],
+                            method=defaults['method'])
+
+  
+  #draw final galaxy image   
+  gal_image = gal.drawImage(nx=defaults['stamp_size'],
+                            ny=defaults['stamp_size'],
+                            scale=defaults['pixel_scale'],
+                            method=defaults['method'])
+  
+  noise = galsim.GaussianNoise()
+  snr = norm.rvs(defaults['mean_snr'],defaults['scatter_snr'],)
+
+  gal_image.addNoiseSNR(noise,snr=snr)
+  
+  #draw kimage of galaxy
+  gal_k =  gs_drawKimage(gal_image.array, 
```

Comment:
here you should explicitly reference your choice of interpolation factor

Reply:
> Well.... Maybe if you want to have the Fourier image of the Galaxy image + noise... Maybe what you have here makes more sense

Yes, I started adding the noise because we moved to store the Fourier images. Ideally this should be outside, but then we should remove galsim from the transform/deconvolution operations, to have everything differentiable.